Project Management Process.md

```diff
@@ -0,0 +1,168 @@
+                    |  
+  Milestones        |     Milestone 1         Milestone A
+    Milestone 1     |       Goal 1              Goal A
+    Milestone 2     |       Goal 2            Milestone B
+    Milestone 3     |       Goal 3              Goal B
+    Milestone A     |     Milestone 2           Goal C
+    Milestone B     |       Goal 4              Goal D
+    Milestone C     |     Milestone 3      
+                    |       Goal 5      
+  Projects          |  
+    Project 1       |   Backlog             Backlog
+    Project 2       |     Goal 1              Goal A
+    Project 3       |     Goal 2              Goal B
+                    |     Goal 3              Goal C
+                    |     Goal 4              Goal D
+                    |     Goal 5
+```
+
+### Goals
+At the heart of each project are the goals. They are the basic unit of work. Goals can be anything that ***adds value to a release***. This includes issues reported by users, feature requests, bugs, Pull-Requests, general tasks such as refactoring, documentation or research. Goals are collected, maintained and tracked in the backlog by the project lead.
+
+A goal description includes the following:
+- Name / Summary
+  - Describes what the goal is on high level
+  - Example 1: "Reduce the size of the go-ipfs executable"
+  - Example 2: "js-ipfs should have API Documentation"
+  - TODO: what is a good name for a goal?
+- Description
+  - What the problem is and how this goal solves it
+- Type
+  - Describe the type of this goal: is it a bug? a new feature? something else?
+- Status
+  - What is the current status of this goal. Is it to-be-reviewed? In progress? Blocked? Done?
+- Dependencies (where applicable)
+- Priority (optional, priority can be defined in the backlog)
+- Any additional information that helps to reach this goal (optional)
+
+TODO: a clear example
+
+### Backlog
+Backlog is the collection of all goals in a project. It should be as comprehensive as possible, including everything that needs to happen or should happen in a particular project. The backlog indicates priority by order: the items that have the highest priority are at the top of the list. The backlog is owned and maintained by the project lead. As part of daily work, the backlog gets updated and a general grooming should happen on weekly basis.
+
+A backlog has the following information:
+- All goals of a project
+- Priority of goals in sorted order
+
+While the backlog is usually one big list of goals, it may sometimes become convoluted. In this case, the backlog list can be split in two: Backlog and Icebox. The icebox section lists goals that will probably not be worked on anytime soon, or are not ready to be worked on yet, whereas the backlog itself contains the goals that are ready or "approved". If split, together they are called "the backlog" and icebox is referenced only in its specific meaning. The split and maintenance of both lists are done the project lead.
```

Comment:
I think we should differentiate "future work" and "backlog" much more explicitly. meaning, "future work" _is not_ part of the backlog. There's three things here:

- 1. Future Work
- 2. "Things that are ready to be worked on" (backlog? Ready?)
- 3. Future Work AND "Things that are ready to be worked on" (backlog? Todos?)

Let's pick a name for both. I think it's typical to call (2.) Backlog in board. but because (1) and (2) were differentiated from (3) (which was historically called Backlog) there's confusion.

I would be in favor of calling (2.) "Ready", which make it very clear that they're ready to go. "backlog" is a more apt name for (2 U 3) i think.